src/buyoutmanager.cpp

```diff
@@ -614,7 +614,7 @@ std::vector<CurrencyType> Currency::Types() {
 }
 
 Currency Currency::FromIndex(int index) {
-    if (index >= currency_type_as_tag_.size()) {
+    if (static_cast<unsigned int>(index) >= currency_type_as_tag_.size()) {
```

Comment:
Use size_t instead if you want to cast, also it needs an additional negative check.